tests/backup/backup_restore_basic_test.go

```diff
@@ -1318,3 +1318,271 @@ var _ = Describe("{BackupSyncBasicTest}", func() {
+		periodicSchedulePolicyUid        string
+		labelForSingleNamespace          string
+		labelForMultipleNamespace        string
+		scheduleName                     string
+		firstScheduleBackupName          string
+		multipleNamespaceBackupName      string
+		scheduleBackupMultipleNamespace  string
+		firstScheduleBackupForMultipleNs string
+		cloudCredUidList                 []string
+		backupNames                      []string
+		restoreNames                     []string
+		bkpNamespaces                    []string
+		scheduleNames                    []string
+		nsLabelsGroup1                   map[string]string
+		nsLabelsGroup2                   map[string]string
+		namespaceMapping                 map[string]string
+		multipleRestoreMapping           map[string]string
+		scheduleRestoreMapping           map[string]string
+		scheduleMultipleRestoreMapping   map[string]string
+		contexts                         []*scheduler.Context
+		appContexts                      []*scheduler.Context
+	)
+	backupLocationMap := make(map[string]string)
+	bkpNamespaces = make([]string, 0)
+	JustBeforeEach(func() {
+		StartTorpedoTest("BasicAndScheduleBackupUsingNamespaceLabel", "Filter using namespace label, perform backup and restore of basic backup, schedule backup of single, multiple namespace", nil, 84842)
+		log.InfoD("Deploy applications")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < 2; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+		log.InfoD("Created namespaces %v", bkpNamespaces)
+	})
+	It("Filter using namespace label, perform backup and restore of basic backup, schedule backup of single, multiple namespace", func() {
+		providers := getProviders()
+		Step("Validate applications", func() {
+			log.InfoD("Validate applications")
+			ValidateApplications(contexts)
+		})
+		Step("Adding labels to namespaces", func() {
+			log.InfoD("Adding labels to namespaces")
+			nsLabelsGroup1 = GenerateRandomLabels(3)
+			err = AddLabelsToMultipleNamespaces(nsLabelsGroup1, bkpNamespaces)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Adding labels [%v] to namespaces [%v]", nsLabelsGroup1, bkpNamespaces))
+			nsLabelsGroup2 = GenerateRandomLabels(3)
+			err = AddLabelsToMultipleNamespaces(nsLabelsGroup1, []string{bkpNamespaces[0]})
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Adding labels [%v] to single namespaces [%v]", nsLabelsGroup2, bkpNamespaces[0]))
+		})
+		Step("Generating namespace label string from label map for single and multiple namespace", func() {
+			log.InfoD("Generating namespace label string from label map for single and multiple namespace")
+			labelForSingleNamespace = mapToString(nsLabelsGroup1)
```

Comment:
i think mismatch in mapping label, As per above step nsLabelsGroup1 contains mutiple namespaces and nsLabelsGroup2 contains single namespace